legend-pure-m3-core/src/main/java/org/finos/legend/pure/m3/tools/PackageTreeIterable.java

```diff
@@ -138,7 +139,11 @@ public boolean hasNext()
         @Override
         public Package next()
         {
-            Package pkg = this.deque.removeFirst();
+            Package pkg = this.deque.pollFirst();
```

Comment:
Im wondering why Sonar is complaining about this? Isnt this exactly the difference between `removeFirst` vs `pollFirst`?

Reply:
I wondered the same thing. But I figure it's just as well to have the throw be explicit.